manifests/init.pp

```diff
@@ -131,11 +132,88 @@
   Boolean                       $use_internet_repo       = false,
   String                        $version                 = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
   String                        $rpm_iteration           = '1',
-  Boolean                       $simp_dashboards         = false
-) inherits ::simp_grafana::params {
+  Boolean                       $simp_dashboards         = false,
+  String                        $admin_pw                = passgen('grafana'),
+  String                        $app_pki_dir             = '/etc/pki/simp_apps/grafana/x509',
+  String                        $app_pki_key             = "${app_pki_dir}/private/${facts['fqdn']}.pem",
+  String                        $app_pki_cert            = "${app_pki_dir}/public/${facts['fqdn']}.pub",
+ ) inherits ::simp_grafana::params {
 
-  $merged_cfg = deep_merge($::simp_grafana::params::cfg, $cfg)
-  $merged_ldap_cfg = deep_merge($::simp_grafana::params::ldap_cfg, $ldap_cfg)
+  # Static defaults
+  $_cfg = {
+    server       => {
+      http_port => 8443,
+      protocol  => 'https',
+      cert_file => $app_pki_cert,
+      cert_key  => $app_pki_key,
+    },
+    security     => {
+      admin_password   => $admin_pw,
+      disable_gravatar => true,
+    },
+    users        => {
+      allow_sign_up    => false,
+      allow_org_create => true,
+      auto_assign_org  => true,
+    },
+    'auth.basic' => { enabled => false },
+    'auth.ldap'  => { enabled => $ldap },
+    #Allows SIMP dashboards to be read from the file system
+    'dashboards.json' => { enabled => true },
+    analytics   => { reporting_enabled => false },
+    snapshot    => { external_enabled => false },
+  }
+
+  if ($ldap) {
+
+    $ldap_group_mapping_defaults = [
+      { group_dn => 'simp_grafana_admins',     org_role => 'Admin'  },
+      { group_dn => 'simp_grafana_editors',    org_role => 'Editor' },
+      { group_dn => 'simp_grafana_editors_ro', org_role => 'Read Only Editor' },
+      { group_dn => 'simp_grafana_viewers',    org_role => 'Viewer' },
+    ]
+
+    $base_dn = simplib::lookup('simp_options::ldap::base_dn', { 'default_value' => simplib::ldap::domain_to_dn() } )
+    $bind_dn = simplib::lookup('simp_options::ldap::bind_dn', { 'default_value' => "uid=%s,${base_dn}" } )
+    $bind_pw = simplib::lookup('simp_options::ldap::bind_pw', { 'default_value' => undef } )
+
+    $ldap_urls   = simplib::lookup('simp_options::ldap::uri', { 'default_value' => [''] } )
+    $ldap_url    = $ldap_urls[0]
```

Comment:
Use the last one by default (-1). It's been a (very loose) convention that the last one is generally the closest one to your server.